hw/top_earlgrey/ip/rstmgr/data/autogen/rstmgr.hjson

```diff
@@ -481,7 +482,8 @@
         A bit vector of all the errors that have occurred in reset manager
       ''',
       swaccess: "rw1c",
-      hwaccess: "hwo",
+      hwaccess: "hrw",
+      async: "clk_esc_i",
```

Comment:
hmmmm.... this is a bit weird to me.
I'm not sure what's going to happen.

This means that on life cycle reset, the `clk_esc_i` portion of things will reset, but the side facing the bus does not.
This might get rid of the fault...but the values are still in there and could lead to problems (i think).

Would it be better to first create a test that would showcase this error at top level and then see? `clk_esc` also might not be the best name, maybe...`clk_alert` or something?

Reply:
even if this async idea doesn't completely work.. it's a good idea.
Maybe we should consider exposing the primary reset to each register as well...and not just use async. 
Ie, it's still synchronous, but it just uses a different reset.

Reply:
Yes, I agree with generating a waveform that shows this error at top level and check hot it works. Do we have a corresponding test or we need to create a new case? Then, I will try a different reset with synchronous regs. 

The clock name will be updated in the next fix.

Reply:
we can check with @matutem , otherwise we may need to quickly make one :)

Reply:
ah yes. regarding point 1. If you look at how the `prim_reg_cdc` is currently implemented, there are two storage components. One that's the real async register, and another that is the bus facing register. The bus facing register uses the same `clk_i` and `rst_ni` of the module. So if the latter is using a por reset, then that means an LC reset would only reset the async register, and not the bus facing one.  This could lead to them being out of sync, that's the main reason I thought it might not completely work. You might read back the value and still get the pre-reset value.

regarding point 2, that's currently not available, but we could consider adding that as a feature into the reggen tool. 

good point on point 3, i think it MIGHT be level. A lot of the fatal errors are from things like the FSM being in the wrong state, so those errors will persist unless you also reset the source of the error.

Reply:
Oh I see. Thanks for info!
I manually updated and uploaded RTL based on sync regs for further review. 

For point 3, if those errors persist regardless of rst_lc_n, I think they may not be handled by clearing regs using rst_lc_n. Instead, we need to fix a root cause of continuous errors.

If they can be caused by level from a state, how about changing the error signaling from level to pulse? Otherwise, the system can struggle with chattering due to continuous errors from one error unless we fix the root cause.

Reply:
yeah i personally think the root cause may need to be fixed, ie... move the error generating logic and the flop into the lc reset domain.  A pulse for a fatal error might not be the best idea, because it implies that if attacked, it only needs to be defeated at a single point in time.  I don't remember enough of the design to say for sure though.

Reply:
I see. For pwrmgr, fault_status errors are already generated in the lc reset domain along with corresponding registers.
But for rstmgr, rstmgr_leaf_rst on the lc domain are connected to rst_lc_src. Only 4 por rst (por_io, por_io_div2, por_io_div4, por_usb) instances are connected to rst_por_aon_n. Do we need to move these 4 instances from rst_por to rst_lc?